python/artm/artm_model.py

```diff
@@ -1011,6 +1023,185 @@ def __repr__(self):
         return 'artm.ARTM(num_topics={0}, num_tokens={1}{2})'.format(
             self.num_topics, num_tokens, class_ids)
 
+    def dump_artm_model(self, data_path):
+        """
+        :Description: dump all necessary model files into given folder.
+
+        :param str data_path: full path to folder (should unexist)
+        """
+        if os.path.exists(data_path):
+            raise IOError('Folder {} already exists'.format(data_path))
+
+        os.mkdir(data_path)
+        # save core score tracker
+        self._master.export_score_tracker(os.path.join(data_path, SCORE_TRACKER_FILENAME))
+        # save phi and n_wt matrices
+        self._master.export_model(self.model_pwt, os.path.join(data_path, PWT_FILENAME))
+        self._master.export_model(self.model_nwt, os.path.join(data_path, NWT_FILENAME))
+        # save theta if has theta_name
+        if self.theta_name is not None:
+            self._master.export_model(self.theta_name, os.path.join(data_path, PTD_FILENAME))
+
+        # save parameters in human-readable format
+        params = {}
+        params['version'] = self.library_version
+        params['creation_time'] = str(datetime.datetime.now())
+        params['num_processors'] = self._num_processors
+        params['cache_theta'] = self._cache_theta
+        params['num_document_passes'] = self._num_document_passes
+        params['reuse_theta'] = self._reuse_theta
+        params['theta_columns_naming'] = self._theta_columns_naming
+        params['seed'] = self._seed
+        params['show_progress_bars'] = self._show_progress_bars
+        params['topic_names'] = self._topic_names
+        params['class_ids'] = self._class_ids
+        params['model_pwt'] = self._model_pwt
+        params['model_nwt'] = self._model_nwt
+        params['theta_name'] = self._theta_name
+        params['synchronizations_processed'] = self._synchronizations_processed
+        params['num_online_processed_batches'] = self._num_online_processed_batches
+        params['initialized'] = self._initialized
+
+        regularizers = {}
+        for name, regularizer in iteritems(self._regularizers.data):
+            tau = None
+            gamma = None
+            try:
+                tau = score.tau
```

Comment:
1) fix it
2) write test for human-readable part.
3) catch only specific exception in all try-catch blocks (10 places)